wandb/filesync/dir_watcher.py

```diff
@@ -111,11 +111,13 @@ def __init__(
         save_name: SaveName,
         file_pusher: FilePusher,
         *args,
+        clock_for_testing: Callable[[], float] = time.time,
```

Comment:
this is interesting... I wonder if we can mock it somehow...

Reply:
Like, with `monkeypatch.setattr(dir_watcher, 'time', Mock())`? (It's only used for `time.time` in two places, so I expect that would work at least for the current code.)

Reply:
I guess I went for dependency-injection instead of monkeypatching because it's more predictable. Monkeypatching, there's some danger that you won't catch _all_ the indirect dependencies of `dir_watcher` on the thing you're patching out. But sure, it appears to be the convention in this codebase, so I'll follow it! Thanks for pointing it out.

Reply:
No... that's fine to have it the other way around also `time.time` patching might cause unexpected consequences... just wanted to understand your reasoning. I wonder if we do injection if we can do it more generic